app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -1336,6 +1337,19 @@ class BrowserFragment : WebFragment(), LifecycleObserver, View.OnClickListener,
             hideFindInPage()
         }
 
+        fun bringURLTLDIntoView() {
+            if (urlView != null) {
+                val paint = urlView!!.paint
+                val charWidth = paint.measureText("a", 0, 1)
+                val urlViewWidth = urlView!!.measuredWidth
+                val numChars = urlViewWidth / charWidth
+                if (url.length > numChars) {
+                    val index = UrlUtils.getIndexOfTLD(url)
+                    urlView?.bringPointIntoView(index)
+                }
+            }
```

Comment:
Doing the `null` check and still having to use `!!` is not super nice. One of the extension methods (maybe `let` or `apply`) may make that nicer.

Another alternative may be "assigning and early return":
```Kotlin
val view = urlView ?: return

// And then use `view`
```